### ISSUE TITLE
Can't link with TF Lite C API static lib - undefined reference to `tflite::DefaultErrorReporter()'

### ISSUE THREAD

**Issue by DanijelDomazet** [tensorflow/tensorflow#55344]
I've built the TensorFlow Lite **static** library, the **C API** version. Cloned [the sources][1] and called CMake while turning OFF the shared build flag (`DTFLITE_C_BUILD_SHARED_LIBS:BOOL=OFF`):  
```
cmake -S ../tensorflow_src/tensorflow/lite/c -DTFLITE_C_BUILD_SHARED_LIBS:BOOL=OFF
cmake --build . -j
```
This built the `libtensorflowlite_c.a`. 

I also linked all other `.a` sub-dependencies from the `_dep` directory into a single `libtensorflowdep.a` library (this includes `absl`, `clog` and all other transient dependencies listed here in the [TensorFlow CMake file][2]).

I tried to test the lib using a minimal example: 
```C
#include <stdio.h>
#include "tensorflow/lite/c/c_api.h"

int main(void) {
	printf("Hello!");

	const char *version = TfLiteVersion();
	printf("Hello from TensorFlow Lite C API library version = %s\n", version);

	return 0;
}
```

`gcc -L"/home/joedoe/out/subprojects/tensorflow" -o "MAIN"  ./src/MAIN.o   -ltensorflowlite_c -ltensorflowdep`

But linking fails:

```
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteModelCreate':
c_api.cc:(.text+0x210): undefined reference to `tflite::DefaultErrorReporter()'
/usr/bin/ld: c_api.cc:(.text+0x22f): undefined reference to `tflite::FlatBufferModel::VerifyAndBuildFromBuffer(char const*, unsigned long, tflite::TfLiteVerifier*, tflite::ErrorReporter*)'
/usr/bin/ld: c_api.cc:(.text+0x276): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteModelCreateFromFile':
c_api.cc:(.text+0x322): undefined reference to `tflite::DefaultErrorReporter()'
/usr/bin/ld: c_api.cc:(.text+0x33d): undefined reference to `tflite::FlatBufferModel::VerifyAndBuildFromFile(char const*, tflite::TfLiteVerifier*, tflite::ErrorReporter*)'
/usr/bin/ld: c_api.cc:(.text+0x384): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteModelDelete':
c_api.cc:(.text+0x433): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteInterpreterOptionsCreate':
c_api.cc:(.text+0x452): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteInterpreterOptionsDelete':
c_api.cc:(.text+0x4a7): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteInterpreterCreate':
c_api.cc:(.text+0x560): undefined reference to `tflite::CreateOpResolver()'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteInterpreterDelete':
c_api.cc:(.text+0x5fa): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteInterpreterResizeInputTensor':
c_api.cc:(.text+0x75a): undefined reference to `tflite::Interpreter::ResizeInputTensor(int, std::vector<int, std::allocator<int> > const&)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteInterpreterAllocateTensors':
c_api.cc:(.text+0x7ee): undefined reference to `tflite::Interpreter::AllocateTensors()'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `TfLiteInterpreterInvoke':
c_api.cc:(.text+0x824): undefined reference to `tflite::delegates::InterpreterUtils::InvokeWithCPUFallback(tflite::Interpreter*)'
/usr/bin/ld: c_api.cc:(.text+0x83e): undefined reference to `tflite::Interpreter::Invoke()'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `tflite::internal::InterpreterCreateWithOpResolver(TfLiteModel const*, TfLiteInterpreterOptions const*, tflite::MutableOpResolver*)':
c_api.cc:(.text+0xaf5): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: c_api.cc:(.text+0xb5b): undefined reference to `tflite::MutableOpResolver::AddAll(tflite::MutableOpResolver const&)'
/usr/bin/ld: c_api.cc:(.text+0xbf1): undefined reference to `tflite::DefaultErrorReporter()'
/usr/bin/ld: c_api.cc:(.text+0xc2f): undefined reference to `tflite::InterpreterBuilder::InterpreterBuilder(tflite::Model const*, tflite::OpResolver const&, tflite::ErrorReporter*)'
/usr/bin/ld: c_api.cc:(.text+0xc53): undefined reference to `tflite::InterpreterBuilder::operator()(std::unique_ptr<tflite::Interpreter, std::default_delete<tflite::Interpreter> >*)'
/usr/bin/ld: c_api.cc:(.text+0xca7): undefined reference to `tflite::Interpreter::SetNumThreads(int)'
/usr/bin/ld: c_api.cc:(.text+0xcd0): undefined reference to `tflite::NnApiDelegate()'
/usr/bin/ld: c_api.cc:(.text+0xcdb): undefined reference to `tflite::Interpreter::ModifyGraphWithDelegate(TfLiteDelegate*)'
/usr/bin/ld: c_api.cc:(.text+0xd88): undefined reference to `tflite::Interpreter::ModifyGraphWithDelegate(TfLiteDelegate*)'
/usr/bin/ld: c_api.cc:(.text+0xe08): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: c_api.cc:(.text+0xe67): undefined reference to `tflite::InterpreterBuilder::~InterpreterBuilder()'
/usr/bin/ld: c_api.cc:(.text+0xebe): undefined reference to `tflite::InterpreterBuilder::~InterpreterBuilder()'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `(anonymous namespace)::CallbackOpResolver::~CallbackOpResolver()':
c_api.cc:(.text+0xf62): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `(anonymous namespace)::CallbackErrorReporter::~CallbackErrorReporter()':
c_api.cc:(.text+0xfba): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro+0x8): undefined reference to `vtable for __cxxabiv1::__si_class_type_info'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro+0x20): undefined reference to `vtable for __cxxabiv1::__si_class_type_info'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `tflite::ErrorReporter::~ErrorReporter()':
c_api.cc:(.text._ZN6tflite13ErrorReporterD0Ev[_ZN6tflite13ErrorReporterD5Ev]+0x24): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `tflite::OpResolver::~OpResolver()':
c_api.cc:(.text._ZN6tflite10OpResolverD0Ev[_ZN6tflite10OpResolverD5Ev]+0x24): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `tflite::MutableOpResolver::MutableOpResolver()':
c_api.cc:(.text._ZN6tflite17MutableOpResolverC2Ev[_ZN6tflite17MutableOpResolverC5Ev]+0x1f): undefined reference to `vtable for tflite::MutableOpResolver'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::default_delete<tflite::FlatBufferModel>::operator()(tflite::FlatBufferModel*) const':
c_api.cc:(.text._ZNKSt14default_deleteIN6tflite15FlatBufferModelEEclEPS1_[_ZNKSt14default_deleteIN6tflite15FlatBufferModelEEclEPS1_]+0x22): undefined reference to `tflite::FlatBufferModel::~FlatBufferModel()'
/usr/bin/ld: c_api.cc:(.text._ZNKSt14default_deleteIN6tflite15FlatBufferModelEEclEPS1_[_ZNKSt14default_deleteIN6tflite15FlatBufferModelEEclEPS1_]+0x2a): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `void std::vector<TfLiteDelegate*, std::allocator<TfLiteDelegate*> >::_M_realloc_insert<TfLiteDelegate* const&>(__gnu_cxx::__normal_iterator<TfLiteDelegate**, std::vector<TfLiteDelegate*, std::allocator<TfLiteDelegate*> > >, TfLiteDelegate* const&)':
c_api.cc:(.text._ZNSt6vectorIP14TfLiteDelegateSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_[_ZNSt6vectorIP14TfLiteDelegateSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_]+0x267): undefined reference to `__cxa_begin_catch'
/usr/bin/ld: c_api.cc:(.text._ZNSt6vectorIP14TfLiteDelegateSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_[_ZNSt6vectorIP14TfLiteDelegateSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_]+0x2d0): undefined reference to `__cxa_rethrow'
/usr/bin/ld: c_api.cc:(.text._ZNSt6vectorIP14TfLiteDelegateSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_[_ZNSt6vectorIP14TfLiteDelegateSaIS1_EE17_M_realloc_insertIJRKS1_EEEvN9__gnu_cxx17__normal_iteratorIPS1_S3_EEDpOT_]+0x2dc): undefined reference to `__cxa_end_catch'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `tflite::MutableOpResolver::~MutableOpResolver()':
c_api.cc:(.text._ZN6tflite17MutableOpResolverD2Ev[_ZN6tflite17MutableOpResolverD5Ev]+0x13): undefined reference to `vtable for tflite::MutableOpResolver'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `tflite::MutableOpResolver::~MutableOpResolver()':
c_api.cc:(.text._ZN6tflite17MutableOpResolverD0Ev[_ZN6tflite17MutableOpResolverD5Ev]+0x24): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::default_delete<tflite::Interpreter>::operator()(tflite::Interpreter*) const':
c_api.cc:(.text._ZNKSt14default_deleteIN6tflite11InterpreterEEclEPS1_[_ZNKSt14default_deleteIN6tflite11InterpreterEEclEPS1_]+0x22): undefined reference to `tflite::Interpreter::~Interpreter()'
/usr/bin/ld: c_api.cc:(.text._ZNKSt14default_deleteIN6tflite11InterpreterEEclEPS1_[_ZNKSt14default_deleteIN6tflite11InterpreterEEclEPS1_]+0x2a): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::vector<TfLiteDelegate*, std::allocator<TfLiteDelegate*> >::_M_check_len(unsigned long, char const*) const':
c_api.cc:(.text._ZNKSt6vectorIP14TfLiteDelegateSaIS1_EE12_M_check_lenEmPKc[_ZNKSt6vectorIP14TfLiteDelegateSaIS1_EE12_M_check_lenEmPKc]+0x5f): undefined reference to `std::__throw_length_error(char const*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::vector<int, std::allocator<int> >::_S_check_init_len(unsigned long, std::allocator<int> const&)':
c_api.cc:(.text._ZNSt6vectorIiSaIiEE17_S_check_init_lenEmRKS0_[_ZNSt6vectorIiSaIiEE17_S_check_init_lenEmRKS0_]+0x62): undefined reference to `std::__throw_length_error(char const*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<int>::deallocate(int*, unsigned long)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIiE10deallocateEPim[_ZN9__gnu_cxx13new_allocatorIiE10deallocateEPim]+0x20): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<std::function<std::unique_ptr<TfLiteDelegate, void (*)(TfLiteDelegate*)> (int)> >::deallocate(std::function<std::unique_ptr<TfLiteDelegate, void (*)(TfLiteDelegate*)> (int)>*, unsigned long)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorISt8functionIFSt10unique_ptrI14TfLiteDelegatePFvPS3_EEiEEE10deallocateEPS9_m[_ZN9__gnu_cxx13new_allocatorISt8functionIFSt10unique_ptrI14TfLiteDelegatePFvPS3_EEiEEE10deallocateEPS9_m]+0x20): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<tflite::FlatBufferModel*>(tflite::FlatBufferModel*)':
c_api.cc:(.text._ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC2IPN6tflite15FlatBufferModelEEET_[_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC5IPN6tflite15FlatBufferModelEEET_]+0x26): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: c_api.cc:(.text._ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC2IPN6tflite15FlatBufferModelEEET_[_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC5IPN6tflite15FlatBufferModelEEET_]+0x4d): undefined reference to `__cxa_begin_catch'
/usr/bin/ld: c_api.cc:(.text._ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC2IPN6tflite15FlatBufferModelEEET_[_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC5IPN6tflite15FlatBufferModelEEET_]+0x5e): undefined reference to `tflite::FlatBufferModel::~FlatBufferModel()'
/usr/bin/ld: c_api.cc:(.text._ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC2IPN6tflite15FlatBufferModelEEET_[_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC5IPN6tflite15FlatBufferModelEEET_]+0x66): undefined reference to `operator delete(void*)'
/usr/bin/ld: c_api.cc:(.text._ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC2IPN6tflite15FlatBufferModelEEET_[_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC5IPN6tflite15FlatBufferModelEEET_]+0x6b): undefined reference to `__cxa_rethrow'
/usr/bin/ld: c_api.cc:(.text._ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC2IPN6tflite15FlatBufferModelEEET_[_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC5IPN6tflite15FlatBufferModelEEET_]+0x77): undefined reference to `__cxa_end_catch'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::~_Sp_counted_base()':
c_api.cc:(.text._ZNSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EED0Ev[_ZNSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EED5Ev]+0x24): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<tflite::OpResolver const*>::deallocate(tflite::OpResolver const**, unsigned long)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIPKN6tflite10OpResolverEE10deallocateEPS4_m[_ZN9__gnu_cxx13new_allocatorIPKN6tflite10OpResolverEE10deallocateEPS4_m]+0x20): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<TfLiteDelegate*>::deallocate(TfLiteDelegate**, unsigned long)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIP14TfLiteDelegateE10deallocateEPS2_m[_ZN9__gnu_cxx13new_allocatorIP14TfLiteDelegateE10deallocateEPS2_m]+0x20): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIiE8allocateEmPKv[_ZN9__gnu_cxx13new_allocatorIiE8allocateEmPKv]+0x30): undefined reference to `std::__throw_bad_alloc()'
/usr/bin/ld: c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIiE8allocateEmPKv[_ZN9__gnu_cxx13new_allocatorIiE8allocateEmPKv]+0x40): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<TfLiteDelegate*>::allocate(unsigned long, void const*)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIP14TfLiteDelegateE8allocateEmPKv[_ZN9__gnu_cxx13new_allocatorIP14TfLiteDelegateE8allocateEmPKv]+0x30): undefined reference to `std::__throw_bad_alloc()'
/usr/bin/ld: c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIP14TfLiteDelegateE8allocateEmPKv[_ZN9__gnu_cxx13new_allocatorIP14TfLiteDelegateE8allocateEmPKv]+0x40): undefined reference to `operator new(unsigned long)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<std::__detail::_Hash_node_base*>::deallocate(std::__detail::_Hash_node_base**, unsigned long)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorIPNSt8__detail15_Hash_node_baseEE10deallocateEPS3_m[_ZN9__gnu_cxx13new_allocatorIPNSt8__detail15_Hash_node_baseEE10deallocateEPS3_m]+0x20): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int>::~pair()':
c_api.cc:(.text._ZNSt4pairINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEiED2Ev[_ZNSt4pairINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEiED5Ev]+0x18): undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<std::__detail::_Hash_node<std::pair<std::pair<tflite::BuiltinOperator, int> const, TfLiteRegistration>, true> >::deallocate(std::__detail::_Hash_node<std::pair<std::pair<tflite::BuiltinOperator, int> const, TfLiteRegistration>, true>*, unsigned long)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorINSt8__detail10_Hash_nodeISt4pairIKS3_IN6tflite15BuiltinOperatorEiE18TfLiteRegistrationELb1EEEE10deallocateEPSA_m[_ZN9__gnu_cxx13new_allocatorINSt8__detail10_Hash_nodeISt4pairIKS3_IN6tflite15BuiltinOperatorEiE18TfLiteRegistrationELb1EEEE10deallocateEPSA_m]+0x20): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `__gnu_cxx::new_allocator<std::__detail::_Hash_node<std::pair<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> const, TfLiteRegistration>, true> >::deallocate(std::__detail::_Hash_node<std::pair<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> const, TfLiteRegistration>, true>*, unsigned long)':
c_api.cc:(.text._ZN9__gnu_cxx13new_allocatorINSt8__detail10_Hash_nodeISt4pairIKS3_INSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEiE18TfLiteRegistrationELb1EEEE10deallocateEPSE_m[_ZN9__gnu_cxx13new_allocatorINSt8__detail10_Hash_nodeISt4pairIKS3_INSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEiE18TfLiteRegistrationELb1EEEE10deallocateEPSE_m]+0x20): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::_Sp_counted_ptr<tflite::FlatBufferModel*, (__gnu_cxx::_Lock_policy)2>::~_Sp_counted_ptr()':
c_api.cc:(.text._ZNSt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EED0Ev[_ZNSt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EED5Ev]+0x24): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTVN6tflite10OpResolverE[_ZTVN6tflite10OpResolverE]+0x10): undefined reference to `__cxa_pure_virtual'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTVN6tflite10OpResolverE[_ZTVN6tflite10OpResolverE]+0x18): undefined reference to `__cxa_pure_virtual'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTVN6tflite13ErrorReporterE[_ZTVN6tflite13ErrorReporterE]+0x20): undefined reference to `__cxa_pure_virtual'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTVSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE[_ZTVSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE]+0x20): undefined reference to `__cxa_pure_virtual'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTVSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE[_ZTVSt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE]+0x30): undefined reference to `__cxa_pure_virtual'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTISt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EE[_ZTISt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EE]+0x0): undefined reference to `vtable for __cxxabiv1::__si_class_type_info'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTIN6tflite10OpResolverE[_ZTIN6tflite10OpResolverE]+0x0): undefined reference to `vtable for __cxxabiv1::__class_type_info'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTIN6tflite13ErrorReporterE[_ZTIN6tflite13ErrorReporterE]+0x0): undefined reference to `vtable for __cxxabiv1::__class_type_info'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTISt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE[_ZTISt16_Sp_counted_baseILN9__gnu_cxx12_Lock_policyE2EE]+0x0): undefined reference to `vtable for __cxxabiv1::__si_class_type_info'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o): in function `std::_Sp_counted_ptr<tflite::FlatBufferModel*, (__gnu_cxx::_Lock_policy)2>::_M_dispose()':
c_api.cc:(.text._ZNSt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv[_ZNSt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv]+0x22): undefined reference to `tflite::FlatBufferModel::~FlatBufferModel()'
/usr/bin/ld: c_api.cc:(.text._ZNSt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv[_ZNSt15_Sp_counted_ptrIPN6tflite15FlatBufferModelELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv]+0x2a): undefined reference to `operator delete(void*)'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.ro._ZTISt11_Mutex_baseILN9__gnu_cxx12_Lock_policyE2EE[_ZTISt11_Mutex_baseILN9__gnu_cxx12_Lock_policyE2EE]+0x0): undefined reference to `vtable for __cxxabiv1::__class_type_info'
/usr/bin/ld: /home/joedoe/out/subprojects/tensorflow/libtensorflowlite_c.a(c_api.cc.o):(.data.rel.local.DW.ref.__gxx_personality_v0[DW.ref.__gxx_personality_v0]+0x0): undefined reference to `__gxx_personality_v0'
collect2: error: ld returned 1 exit status
make: *** [makefile:45: TF] Error 1
```

  [1]: https://github.com/tensorflow/tensorflow
  [2]: https://github.com/tensorflow/tensorflow/blob/e41d9eb9edcc1ff2cf52736edde76e6f5d62f6ed/tensorflow/lite/CMakeLists.txt#L139

**Comment by DanijelDomazet** [comment]
@sushreebarsa Any idea? Thanks.

**Comment by mohantym** [comment]
Hi @DanijelDomazet !  Attaching relevant threads for reference. [1](https://github.com/tensorflow/tensorflow/issues/36661#issuecomment-631975005) , [2](https://github.com/tensorflow/tensorflow/issues/36661#issuecomment-585054685) . Can you also switch to g++ instead of gcc and let us know the results? . Thank you!

**Comment by liupeiqiHN** [comment]
@mohantym but how to fix this error on MSVC?

**Comment by DanijelDomazet** [comment]
> Hi @DanijelDomazet ! Attaching relevant threads for reference. [1](https://github.com/tensorflow/tensorflow/issues/36661#issuecomment-631975005) , [2](https://github.com/tensorflow/tensorflow/issues/36661#issuecomment-585054685) . Can you also switch to g++ instead of gcc and let us know the results? Thank you!

Hi, 
- [1](https://github.com/tensorflow/tensorflow/issues/36661#issuecomment-631975005) didn't help as it is about shared `.so` lib, and I am trying to build the static lib. 
- tried `g++`, but it too didn't help

So still can't build the static library for C API.

**Comment by feliwir** [comment]
Is there maybe a way to make the target `tensorflowdep` into upstream?

**Comment by aaronhsueh0506** [comment]
Hi, 
I have same issue on ubuntu 18.04.

I try to follow the command on tensorflow.org
`cmake ../tensorflow_src/tensorflow/lite/c`
`cmake --build . -j`
And I get a libtensorflow_c.so in tflite_build folder.

But when I compile my main.cpp in my project, the error message like this issue, lots of undenfined reference.

Is this have any update?

**Comment by pkgoogle** [comment]
Hi @DanijelDomazet,

Thanks for reporting the issue. I was able to follow these steps and build the static library successfully:
```
cmake -S ../tensorflow_src/tensorflow/lite/c -DTFLITE_C_BUILD_SHARED_LIBS:BOOL=OFF
cmake --build . -j
```

It seems you are making these commands from a sibling directory of the source directory. It is hard to follow exactly what you are doing which may change things quite a bit. i.e. what exactly do you mean by this? 

"I also linked all other .a sub-dependencies from the _dep directory into a single libtensorflowdep.a library (this includes absl, clog and all other transient dependencies listed here in the [TensorFlow CMake file](https://github.com/tensorflow/tensorflow/blob/e41d9eb9edcc1ff2cf52736edde76e6f5d62f6ed/tensorflow/lite/CMakeLists.txt#L139))."

Also what is the filename of your test script, it looks like it should be MAIN.cpp but that information is implied. To help us, help you the best can you follow the instructions here: https://github.com/tensorflow/tensorflow/blob/master/ISSUE_TEMPLATE.md and for exact commands please give us context on your directory structure as well. Think of it as how you would write a shell script. Thanks for your help

**Comment by DanijelDomazet** [comment]
@pkgoogle Thanks, but a bit late now, I reported the issue more than a year ago. It might as well work properly by now. However, I'm not using TensorFlow C Lite at the moment in my projects, so can't check it. Maybe someone else can. Thanks for your effort.
